routing.go

```diff
@@ -445,18 +444,13 @@ func (dht *IpfsDHT) Provide(ctx context.Context, key cid.Cid, brdcst bool) (err
 		return err
 	}
 
-	mes, err := dht.makeProvRecord(keyMH)
-	if err != nil {
-		return err
-	}
-
 	wg := sync.WaitGroup{}
 	for p := range peers {
 		wg.Add(1)
 		go func(p peer.ID) {
 			defer wg.Done()
 			logger.Debugf("putProvider(%s, %s)", keyMH, p)
-			err := dht.sendMessage(ctx, p, mes)
+			err := dht.protoMessenger.PutProvider(ctx, p, keyMH, dht.host)
```

Comment:
This is slightly less efficient since we create the record `len(peers)` times. If that's an issue we can make a function that allows that, but this seems likely to be fine. Thoughts?